Bundler/Viewer/Unity/Assets/Scripts/Network/Scripts/NetworkAnchorManager.cs

```diff
@@ -0,0 +1,487 @@
+    {
+        if (completeCallback != null)
+        {
+            completeCallback(SerializationCompletionReason.NotSupported, null);
+        }
+    }
+}
+#endif
+
+/// <summary>
+/// Data describing the source of the shared anchor data.
+/// </summary>
+[Serializable]
+public struct SharedAnchorData
+{
+    public static SharedAnchorData Create(string anchorId)
+    {
+        SharedAnchorData result;
+        result.SourceIp = NetworkAnchorManager.Instance == null ? "unknown" : NetworkAnchorManager.Instance.LocalAddress;
+        result.AnchorId = anchorId;
+        return result;
+    }
+
+    /// <summary>
+    /// The IP address of the player that is serving the most recent anchor
+    /// </summary>
+    public string SourceIp;
+
+    /// <summary>
+    /// The id (or name) of the anchor.
+    /// </summary>
+    public string AnchorId;
+
+    /// <summary>
+    /// Test if this meta data object hold valid information
+    /// </summary>
+    public bool IsValid
+    {
+        get
+        {
+            return !String.IsNullOrEmpty(SourceIp) && !String.IsNullOrEmpty(AnchorId);
+        }
+    }
+
+    public override string ToString()
+    {
+        return String.Format("(anchor source ip: {0}) (anchor id: {1})", SourceIp, AnchorId);
+    }
+}
+
+public class LastReceivedAnchorArgs
+{
+    public LastReceivedAnchorArgs(WorldAnchorTransferBatch transferBatch)
+    {
+        TransferBatch = transferBatch;
+    }
+
+    /// <summary>
+    /// The transfer batch containing the anchor
+    /// </summary>
+    public WorldAnchorTransferBatch TransferBatch { get; private set; }
+
+    /// <summary>
+    /// The first anchor id in the transfer batch
+    /// </summary>
+    public string AnchorId
+    {
+        get
+        {
+            if (TransferBatch == null)
+            {
+                return String.Empty;
+            }
+
+            var ids = TransferBatch.GetAllIds();
+            if (ids == null || ids.Length == 0)
+            {
+                return String.Empty;
+            }
+
+            return ids[0];
+        }
+    }
+}
+
+public class NetworkAnchorManager : NetworkBehaviour
+{
+    /// <summary>
+    /// Get the local instance of the anchor manager
+    /// </summary>
+    private static NetworkAnchorManager instance;
+    public static NetworkAnchorManager Instance
+    {
+        get
+        {
+            return instance;
+        }
+    }
+
+    /// <summary>
+    /// This event is raised when a new anchor arrives from a different player.
+    /// </summary>
+    /// <param name="args">Contains the data that arrived.</param>
+    public delegate void OnLastReceivedAnchor(NetworkAnchorManager sender, LastReceivedAnchorArgs args);
+    public event OnLastReceivedAnchor LastReceivedAnchorChanged;
+
+    /// <summary>
+    /// Get if the local anchor player is in the process of receiving a shared anchor.
+    /// </summary>
+    public bool LoadingAnchor
+    {
```

Comment:
"{ get; private set; }" has been used elsewhere in this PR. Why the multi-line style here?

Reply:
"fixed"  I'm new, and was told there weren't any style guidelines :)